homeassistant/components/tado/repairs.py

```diff
@@ -17,10 +17,9 @@ def manage_water_heater_fallback_issue(
     integration_overlay_fallback: str | None,
 ) -> None:
     """Notify users about water heater respecting fallback setting."""
-    if (
-        integration_overlay_fallback
-        in [CONST_OVERLAY_TADO_DEFAULT, CONST_OVERLAY_MANUAL]
-        and len(water_heater_entities) > 0
+    if integration_overlay_fallback in (
+        CONST_OVERLAY_TADO_DEFAULT,
+        CONST_OVERLAY_MANUAL,
     ):
         for water_heater_entity in water_heater_entities:
```

Comment:
... something like:

```python
for zone_name in water_heater_names:
  ...
      issue_id=f"{WATER_HEATER_FALLBACK_REPAIR}_{zone_name}"
```

So, in **water_heater.py**:
```
    manage_water_heater_fallback_issue(
        hass=hass,
        water_heater_names=(e.zone_name for e in entities),
        integration_overlay_fallback=tado.fallback,
    )
```

Reply:
This weirdly causes (e.zone_name for e in entities), `error: "WaterHeaterEntity" has no attribute "zone_name"  [attr-defined]` error even though the zone_name attribute is available in the TadoZoneEntity parent class.